### PR TITLE
Linear sw

### PR DESCRIPTION
**Pull request by nhartney**
This branch corrects the bug in the implementation of the linear shallow water equations where facet terms were being incorrectly added to the depth transport. It adds a new test for the linear shallow water equations based on the wave scenario from Schreiber, Peixoto, Haut and Wingate (2016), comparing the output to a known checkpoint answer.

### CODE REVIEW

File: integration-tests/equations/test_linear_sw_wave.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    # Equation
+    parameters = ShallowWaterParameters(H=H, g=g)
+    fexpr = Constant(1)
+    eqns = LinearShallowWaterEquations(domain, parameters, fexpr=fexpr)
+
+    # I/O
+    output = OutputParameters(dirname=str(tmpdir)+"/linear_sw_wave",
+                              dumpfreq=1,
+                              log_level='INFO')
+    io = IO(domain, output, diagnostic_fields=[CourantNumber()])
```
Comment (jshipton) [inline integration-tests/equations/test_linear_sw_wave.py:37]: do we need this diagnostic?

**Review by jshipton — CHANGES_REQUESTED** [review]
this looks great - just a couple of minor changes - thanks Nell!

File: integration-tests/equations/test_linear_sw_wave.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+def test_linear_sw_wave(tmpdir):
+
+    dirname = str(tmpdir)
+    stepper, check_stepper = run_linear_sw_wave(dirname)
+
+    for variable in ['u', 'D']:
+        new_variable = stepper.fields(variable)
+        check_variable = check_stepper.fields(variable)
+        error = norm(new_variable - check_variable) / norm(check_variable)
+        print(error)
```
[the remark this answers is not in the document]
Comment (nhartney) [reply integration-tests/equations/test_linear_sw_wave.py]: Ah yes! I forgot to get rid of it after testing.

File: integration-tests/equations/test_linear_sw_wave.py (hunk 1 above)
Comment (nhartney) [reply integration-tests/equations/test_linear_sw_wave.py]: I think we don't.

**Pull request merged**